### PR TITLE
Ets 281 phac cronjob

### PR DESCRIPTION
**Pull request by ashtonmeuser**
Based off of and supersedes #94.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -16,8 +16,10 @@
     "aws-sdk": "2.654.0",
     "axios": "0.19.2",
     "body-parser": "1.19.0",
+    "cron": "1.8.2",
     "crypto": "1.0.1",
     "dayjs": "1.8.27",
+    "es6-promise-pool": "^2.5.0",
```
Comment (mauriciofreshw) [inline server/package.json:22]: Version not locked

Comment (ashtonmeuser) [reply server/package.json:22]: Good catch. Fixed.

File: server/cron-job.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const { CronJob } = require('cron');
+const logger = require('./logger.js');
+const { dbClient, collections } = require('./db');
+const { getUnsuccessfulSbcTransactions, updateSbcTransactions } = require('./utils/sbc-phac-queries');
+const { postServiceItem } = require('./utils/service-bc');
+const { sendPhacToSBC } = require('./utils/sbc-phac');
+
+const postToSbcAndUpdateDb = async (collection, submission) => {
```
Comment (fwkendall) [inline server/cron-job.js:8]: I'm not loving that we now have this in both cron-job.js and in sbc-phac.js. This shouldn't block anything mind you, but feels ripe for refactoring into a shared util both files can reference

Comment (ashtonmeuser) [reply server/cron-job.js:8]: Yes, agreed. Looks like the way the cron jobs were structured, the ETS → SBC job is implemented in `cron-job.js` while I chose to implement the PHAC → SBC job in `sbc-phac.js`. Both ways would work, but should be aligned. I would cast my vote for the latter i.e. abstracting the logic of the job into its own file.

Comment (fwkendall) [reply server/cron-job.js:8]: I'm fine with that approach btw, if this is something you want to tackle today as part of the work you're doing :)

File: server/cron-job.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    } catch (error) {
+      logger.error(`Failed one or more SBC transactions: ${error}`, currentDate);
+    }
+  } catch (error) {
+    logger.error(`Failed to establish DB connection: ${error}`, currentDate);
+  }
+};
+
+const phacToSbcJob = async () => {
+  if (process.env.ENABLE_PHAC_CRONJOB !== 'true') return;
```
Comment (ashtonmeuser) [inline server/cron-job.js:39]: Fail safe. Should be able to remove `DISABLE_PHAC_CRONJOB` from envs if it has already been set.

File: server/utils/sbc-phac.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  return sbcItem;
+};
+
+function* makeSbcTransactionIterator(collection, data) {
+  for (const d of data) { // eslint-disable-line no-restricted-syntax
+    yield postToSbcAndUpdateDb(collection, d);
+  }
+}
+
+const sendPhacToSBC = async () => {
```
Comment (fwkendall) [inline server/utils/sbc-phac.js:90]: it looks like this is actually sending both phac and ets?

**Comment by fwkendall** [comment]
Updates look good, just added one comment about naming :)

**Pull request merged**